src/common/AmazonS3Backend.ts

```diff
@@ -304,11 +304,26 @@ export class AmazonS3Backend implements IBackend {
 
             // aws stores the part md5 in the ETag for multipart uploads
             if (partMD5 === data.Parts[i].ETag) {
-              // TODO add part to completed parts in upload
+
+              // modify current upload object if it exists
+              if (params.Body.s3upload) {
+
+                // ignore part if it has been or is being uploaded
+                if (params.Body.s3upload['completeInfo'][partNumber]) {
```

Comment:
Seeing the following warning:
`[312, 42]: object access via string literals is disallowed`

We can fix the warning by changing this line to `params.Body.s3upload.completeInfo[partNumber]`

Reply:
This fix works for `totalUploadedBytes` and `doneParts`, but causes an error with `completeInfo`.  Any clue as to why that one is different?